Tests/LeafKitTests/TestHelpers.swift

```diff
@@ -0,0 +1,267 @@
+import XCTest
+import NIOConcurrencyHelpers
+@testable import LeafKit
+
+/// Assorted multi-purpose helper pieces for LeafKit tests
+
+// MARK: - Helper Functions
+
+/// Directly run a String "template" through `LeafLexer`
+/// - Parameter str: Raw String holding Leaf template source data
+/// - Returns: A lexed array of LeafTokens
+internal func lex(_ str: String) throws -> [LeafToken] {
+    var lexer = LeafLexer(name: "lex-test", template: str)
+    return try lexer.lex().dropWhitespace()
+}
+
+/// Directly run a String "template" through `LeafLexer` and `LeafParser`
+/// - Parameter str: Raw String holding Leaf template source data
+/// - Returns: A lexed and parsed array of Syntax
+internal func parse(_ str: String) throws -> [Syntax] {
+    var lexer = LeafLexer(name: "alt-parse", template: str)
+    let tokens = try! lexer.lex()
+    var parser = LeafParser(name: "alt-parse", tokens: tokens)
+    let syntax = try! parser.parse()
+
+    return syntax
+}
+
+/// Directly run a String "template" through full render chain
+/// - Parameter template: Raw String holding Leaf template source data
+/// - Parameter context: LeafData context
+/// - Returns: A fully rendered view
+internal func render(name: String = "test-render", _ template: String, _ context: [String: LeafData] = [:]) throws -> String {
+    var lexer = LeafLexer(name: name, template: template)
+    let tokens = try lexer.lex()
+    var parser = LeafParser(name: name, tokens: tokens)
+    let ast = try parser.parse()
+    var serializer = LeafSerializer(
+        ast: ast,
+        context: context
+    )
+    let view = try serializer.serialize()
+    return view.getString(at: view.readerIndex, length: view.readableBytes) ?? ""
+}
+
+// MARK: - Helper Structs and Classes
+
+/// Helper wrapping` LeafRenderer` to preconfigure for simplicity & allow eliding context
```

Comment:
Passthrough wrapper for LeafRenderer to avoid having to always set up everything and cluttering test cases with typical defaults - `file` is the only non-defaulting parameter